src/libraries/Microsoft.Extensions.DependencyModel/src/DependencyContextJsonReader.cs

```diff
@@ -683,18 +680,18 @@ private static List<RuntimeFallbacks> ReadRuntimes(ref Utf8JsonReader reader)
             return runtimeFallbacks;
         }
 
-        private IEnumerable<Library> CreateLibraries(IEnumerable<TargetLibrary> libraries, bool runtime, Dictionary<string, LibraryStub> libraryStubs)
+        private IEnumerable<Library> CreateLibraries(IEnumerable<TargetLibrary>? libraries, bool runtime, Dictionary<string, LibraryStub>? libraryStubs)
         {
             if (libraries == null)
             {
                 return Enumerable.Empty<Library>();
             }
             return libraries
                 .Select(property => CreateLibrary(property, runtime, libraryStubs))
-                .Where(library => library != null);
+                .Where(library => library != null)!;
```

Comment:
Why is this necessary?

Reply:
Unfortunately yes. Compiler doesn't recognize `.Where(library => library != null)` as a null check.

Reply:
Created an issue: https://github.com/dotnet/roslyn/issues/55667

Reply:
I get it now - `CreateLibrary` can return `Library?`, so after the `.Select`, the variable is now `IEnumerable<Library?>`, and then the `Where` doesn't change the enumerable variable's type - it takes in `IEnumerable<Library?>` and returns `IEnumerable<Library?>`.

Thanks for the explanation.